### PR TITLE
- Changed tests to use Fixugen

### PR DESCRIPTION
**Pull request by aatuny**
- Added support for configurable filters
- Fixed CLI to utilize all configuration options

### CODE REVIEW

File: src/transform/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+function callback({getFixture, isLegalDeposit = false, sourceMap = {}, filters = {}}) {
+  const momentMock = () => moment('2020-01-01T00:00:00');
+
+  const inputData = getFixture({components: ['input.json'], reader: READERS.JSON});
+  const expectedRecord = getFixture({components: ['output.json'], reader: READERS.JSON});
+  const expectedError = getFixture('error.txt');
+
+  const transform = createTransformer({
+    isJson: true,
```
Comment (natlibfi-jonollil) [inline src/transform/index.spec.js:53]: isJson could be metadata variable. so in tests could be used some real XML styled data too

**Review by natlibfi-jonollil — CHANGES_REQUESTED** [review]

File: src/transform/convert/convertRecord.js [hunk 2]
```diff
@@ -27,9 +27,7 @@
 */
 
 import {MarcRecord} from '@natlibfi/marc-record';
-import moment from 'moment';
-
-import {sourceMap} from '../../config';
+import momentOrig from 'moment';
```
[the remark this answers is not in the document]
Comment (aatuny) [reply src/transform/convert/convertRecord.js]: Altered original import to use the library name

File: src/transform/index.spec.js (hunk 1 above)
Comment (aatuny) [reply src/transform/index.spec.js]: Changed isJson value to be derived from metadata file if it's been defined there

File: src/transform/index.spec.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
+/**
```
Comment (natlibfi-jonollil) [inline src/transform/index.spec.js:1]: In test-fixture metadata files add key "description" and as its value description of thing that this test tests

File: src/transform/convert/convertRecord.js [hunk 4]
```diff
@@ -43,7 +41,10 @@ import {generate776} from './generate7xxFields';
 import {generate856} from './generate8xxFields';
 import {generateSID} from './generateSidFields';
 
-export default (fieldValueInterface, harvestSource) => {
+export default (fieldValueInterface, options) => {
+  const {harvestSource, sourceMap} = options;
+  const momentSource = options.moment || moment;
+
   const marcRecord = new MarcRecord();
   marcRecord.leader = '01704nam a22002653i 4500'; // eslint-disable-line functional/immutable-data
```
Comment (ammsalme) [inline src/transform/convert/convertRecord.js:49]: 1) Are records always text monographs 'am' - or are these leader values updated later if needed? 
2) Character positions 00-04 - Record length and 12-16 - Base address of data could be all zeroes, as actual values for these are not known yet.

**Review by ammsalme — APPROVED** [review]

File: src/transform/convert/convertRecord.js (hunk 4 above)
Comment (aatuny) [reply src/transform/convert/convertRecord.js]: 1. Good point. At this point I believe this is what the transformation supports, but we need to add filter to make it explicit.
2. Fixed

**Pull request merged**